Source/WPF/MyMoney/Controls/TrendGraph.xaml.cs

```diff
@@ -267,45 +267,56 @@ private TrendGraphSeries AddSeries(DateTime start, DateTime end, Color color)
                 Transaction t = row as Transaction;
                 if (t == null) continue;
 
                 if ( t.Account == this.account || // showing transactions for an account
-                    (account == null )) // showing transactions by category // && ((!t.IsBudgeted && t.Account.IsBudgeted))
+                    (this.account == null )) // showing transactions by category // && ((!t.IsBudgeted && t.Account.IsBudgeted))
                 {
-                    decimal v = t.GetCategorizedAmount(this.category);
+                    // If the trend includes the first transaction, also include the opening balance
+                    if (firstTransaction && this.account != null &&  t.Date >= start)
+                    {
+                        balance = (double)this.account.OpeningBalance;
+                    }
+                    firstTransaction = false;
+
                     if (t.Date >= start && t.Date <= end)
                     {
-                        started = true;
-                        if (first)
+                        // If the transactions are on the same day, don't add to the graph yet.  
+                        // Accumulate them and the accumulated value for the day will be displayed
+                        if (last != t.Date)
                         {
-                            string label = GetLabel(balance, t);
-                            timeData.Add(new ChartValue(label, (double)balance, t));
-                            first = false;
+                            AddDatum(balance, last, t.Date, t, timeData);
                         }
 
-                        balance = AddDatum(balance, last, t.Date, t, timeData, (double)v);
+                        // For all regular transaction lists, we calcuclate the overall balance on the fly, based on each transaction amount.
+                        // When we list securities - not an account - we should show the overall value of the securities instead, 
+                        // which is precalculated and stored within each transaction
+                        if (this.account == null && t.Investment != null)
+                        {
+                            balance = (double) t.RunningBalance;
+                        }
+                        else
+                        {
+                            balance += (double) t.GetCategorizedAmount(this.category);
+                        }
                         last = t.Date;
                         lastt = t;
                     }
-                    if (started)
-                    {
-                        balance += (double)v;
-                    }
                 }
             }
 
-            if (last != end && lastt != null)
+            // Put the last transaction in the list
+            if (lastt != null)
             {
-                balance = AddDatum(balance, last, end, lastt, timeData, 0);
+                AddDatum(balance, last, end.AddDays(1), lastt, timeData);

```

Comment:
I don't understand this or the old code on this line, lastt was set to a transaction added by AddDatum already, so why are we adding it again here anyway?

Reply:
The code did not properly add the last transaction if it was on the last day.

Reply:
Got it, I see now.